### PR TITLE
fix(mdns): Fix compile error when MDNS_PREDEF_NETIF_ETH is defined, but ETH_ENABLED is not (IDFGH-11868)

### PR DESCRIPTION
**Pull request by marekmaskarinec**
This is a fix for the issue #459.

### CODE REVIEW

**Review by david-cermak — COMMENTED** [review]
Thanks for posting the fix!

I think that we shouldn't try to handle Ethernet events if `CONFIG_MDNS_PREDEF_NETIF_ETH=n`. 
Would this change fix the issue?
```patch
diff --git a/components/mdns/mdns.c b/components/mdns/mdns.c
index 1149ff7..2c95bdc 100644
--- a/components/mdns/mdns.c
+++ b/components/mdns/mdns.c
@@ -3925,7 +3925,7 @@ void mdns_preset_if_handle_system_event(void *arg, esp_event_base_t event_base,
             break;
         }
     }
-#if CONFIG_ETH_ENABLED
+#if CONFIG_ETH_ENABLED && CONFIG_MDNS_PREDEF_NETIF_ETH
     else if (event_base == ETH_EVENT) {
         switch (event_id) {
         case ETHERNET_EVENT_CONNECTED:
```

**Comment by marekmaskarinec** [comment]
I didn't do this because I thought the user could add their own
Ethernet netif, even if the predefined one is disabled. Should I add
this?

**Comment by david-cermak** [comment]
> I thought the user could add their own Ethernet netif, even if the predefined one is disabled.

Yes, this is the idea, but if users add their own interfaces, they have to take care of event handling in the application code (as the library couldn't possibly handle status changes of various netifs which are under user control), there's a convenient API provided just for this purpose:

https://github.com/espressif/esp-protocols/blob/5ab699d6f4906a3ce6c97377f798944533d7152e/components/mdns/include/mdns.h#L812-L831

The role of predefined interfaces is a legacy handling of the default netifs that were present (hardcoded) in older IDF versions.

> Should I add this?

If you're willing to address this issue, I'd suggest you update your commit accordingly. (but only if you feel like you'd like to contribute, which is much appreciated :-)

**Comment by marekmaskarinec** [comment]
> I think that we shouldn't try to handle Ethernet events if CONFIG_MDNS_PREDEF_NETIF_ETH=n.

Done.

File: components/mdns/mdns.c [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #include "esp_log.h"
 #include "esp_random.h"
 
-#if CONFIG_ETH_ENABLED && CONFIG_MDNS_PREDEF_NETIF_ETH
+#if CONFIG_ETH_ENABLED
```
Comment (david-cermak) [inline components/mdns/mdns.c:21]: suggest reverting this change

```suggestion
#if CONFIG_ETH_ENABLED && CONFIG_MDNS_PREDEF_NETIF_ETH
```

The reason is that even if we're using a custom Ethernet interface there's no need for `esp_eth` headers (as the netif itself is well abstracted), which helps reduce component coupling.

**Review by david-cermak — COMMENTED** [review]
Thanks for the effort! One minor nitpick, LGTM otherwise.

**Comment by david-cermak** [comment]
Saw that you had pushed from dce4339 to e529bee, but there's no diff in the code.

Also please note that we've got some trouble with our CI, most of the tests are failing (should be fixed soon).
(This [one](https://github.com/espressif/esp-protocols/actions/runs/7503602020/job/20428699989?pr=479) however relies on the Ethernet module dependency -- should be fixed once you address the last comment)

**Comment by marekmaskarinec** [comment]
Sorry, I seem to have committed incorrectly.

**Review by david-cermak — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**